src/index.js

```diff
@@ -9,67 +10,94 @@
  * Module dependencies.
  */
 
-const { isValid: isValidEin, mask: maskEin } = require('ein-validator');
-const { isValid: isValidItin, mask: maskItin } = require('itin-validator');
-const { isValid: isValidSsn, mask: maskSsn } = require('ssn-validator');
+const euTinValidator = require('./validators/eu-tin-validator');
+const usTinValidator = require('./validators/us-tin-validator');
 
 /**
- * Excludes repeated numbers as TIN e.g. 111111111.
+ * Constants.
  */
 
+const defaultCountry = 'US';
+const defaultEntityType = 'natural-person';
+// Excludes ascending sequence as TIN e.g. 123456789.
+const sequence = Array.from({ length: 10 }, (_, i) => i).join('').repeat(2);
+// Excludes repeated numbers as TIN e.g. 111111111.
 const repeatedNumbers = Array.from({ length: 10 }, (_, i) => i).map(current => String(current).repeat(9));
-
-/**
- * Excludes ascending and descending sequence as TIN e.g. 123456789.
- */
-
-const sequence = Array.from({ length: 10 }, (_, i) => i).reduce((acc, current) => acc + current, '').repeat(2);
+// Excludes descending sequence as TIN e.g. 987654321.
 const reverseSequence = sequence.split('').reverse().join('');
 
 /**
- * Export `isValid` function.
+ * Run basic TIN validation.
  */
 
-module.exports.isValid = value => {
+const runBasicValidation = value => {
   const sanitizedValue = value.replace(/\D/g, '');
+  const isRepeatedSequence = repeatedNumbers.includes(sanitizedValue);
 
-  if (repeatedNumbers.indexOf(sanitizedValue) !== -1) {
+  if (isRepeatedSequence) {
     return false;
   }
 
-  if (sequence.includes(sanitizedValue) || reverseSequence.includes(sanitizedValue)) {
+  const isAscendingSequence = sequence.includes(sanitizedValue);
+  const isDescendingSequence = reverseSequence.includes(sanitizedValue);
+
+  if (isAscendingSequence || isDescendingSequence) {
     return false;
   }
 
-  return isValidSsn(value) || isValidItin(value) || isValidEin(value);
+  return true;
 };
 
 /**
- * Export `mask` funtion.
+ * Export `isValid` function.
  */
 
-module.exports.mask = value => {
-  if (!module.exports.isValid(value)) {
-    throw new Error('Invalid Taxpayer Identification Number');
+module.exports.isValid = async (value, { country = defaultCountry, entityType = defaultEntityType } = {}) => {
+  const isValid = runBasicValidation(value);
+
+  if (!isValid) {
+    return false;
   }
 
-  if (isValidEin(value)) {
-    return maskEin(value);
+  if (usTinValidator.isCountrySupported(country)) {
+    return usTinValidator.isValid(value);
   }
 
-  if (isValidItin(value)) {
-    return maskItin(value);
+  if (euTinValidator.isCountrySupported(country)) {
+    return await euTinValidator.isValid(value, { country, entityType });
   }
 
-  if (isValidSsn(value)) {
-    return maskSsn(value);
+  return true;
+};
+
+/**
+ * Export `mask` funtion.
+ */
+
+module.exports.mask = async (value, { country = defaultCountry, entityType = defaultEntityType }) => {
```

Comment:
```suggestion
module.exports.mask = async (value, { country = defaultCountry, entityType = defaultEntityType } = {} ) => {
```

Since this is failing on identity-service
```
> await tinValidator.mask('fooo')
Uncaught TypeError: Cannot read properties of undefined (reading 'country')
    at Object.module.exports.mask (/Users/alexandre.pacheco/Projects/Uphold/identity-service/node_modules/tin-validator/src/index.js:77:39)
    at /Users/alexandre.pacheco/Projects/Uphold/identity-service/<repl>.ts:1:54
    at /Users/alexandre.pacheco/Projects/Uphold/identity-service/<repl>.ts:3:4
    at Script.runInContext (node:vm:149:12)
    at runInContext (/Users/alexandre.pacheco/Projects/Uphold/identity-service/node_modules/ts-node/src/repl.ts:675:19)
    at Object.execCommand (/Users/alexandre.pacheco/Projects/Uphold/identity-service/node_modules/ts-node/src/repl.ts:633:32)
```